yksmp/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::ffi::c_void;
+use std::{ptr, slice};
+
+/// Parses LLVM stackmaps version 3 from a given address. Provides a way to query relevant
+/// locations given the return address of a `__llvm_deoptimize` function.
+/// Note that LLVM currently only supports stackmaps for 64 bit architectures. Once they support
+/// others we will need to either make this parser more dynamic or create a new one for each
+/// architecture.
+struct StackMapParser<'a> {
+    data: &'a [u8],
+    offset: usize,
+}
+
+struct Function {
+    addr: u64,
+    record_count: u64,
+}
+
+struct Record {
+    offset: u32,
+    locs: Vec<Location>,
+    liveouts: Vec<LiveOut>,
+}
+
+#[derive(Debug)]
+enum Location {
+    Register(u16),
+    Direct(u16, i32),
+    Indirect(u16, i32),
+    Constant(u32),
+    LargeConstant(u64),
+}
+
+#[allow(dead_code)]
+struct LiveOut {
+    dwreg: u16,
+    size: u8,
+}
+
+impl StackMapParser<'_> {
```

Comment:
Would be good to move the struct def next to its impl block.

Reply:
Fixed in 8be5b0d.